datasets/_global/tokyo_mou/crawler.py

```diff
@@ -0,0 +1,141 @@
+import httpx
+import re
+
+from lxml import etree
+from lxml import html
+from pprint import pprint
+from time import sleep
+
+from zavod import Context
+
+imo = "8817007"
+MAX_RETRIES = 5
+RETRY_DELAY = 5  # seconds
+
+
+def solve_arithmetic(expression: str) -> str:
+    match = re.search(r"(\d+)\s*([+\-*/])\s*(\d+)", expression)
+    if not match:
+        raise ValueError(f"Invalid CAPTCHA expression: {expression}")
+    a, op, b = match.groups()
+    return str(eval(f"{a}{op}{b}"))
```

Comment:
Let's add a comment to make it super super clear how dangerous this can be.

Can we make the regex more strict by matching start/end with ^ and $?

```suggestion
    # WARNING!! WARNING!!
    # This is validating external input to eval() - review EXTREMELY carefully.
    # Match an expression like 123 + 456
    match = re.search(r"^(\d+)\s*([+\-*/])\s*(\d+)$", expression)
    if not match:
        raise ValueError(f"Invalid CAPTCHA expression: {expression}")
    a, op, b = match.groups()
    return str(eval(f"{a}{op}{b}"))
```

Reply:
I feel like an `if op == "+" ... elif op == "-"` wouldn't hurt that much and make it feel a bit safer :)

Reply:
so much nicer